README.md

```diff
@@ -381,464 +607,224 @@ class MyBaseClass extends WidgetBase<WidgetProperties> {
+The properties which can be set on `DomWrapper` are the combination of the `WidgetBaseProperties` and the `VirtualDomProperties`, which means effectively you can use any of the properties passed to a `v()` node and they will be applied to the wrapped DOM node.  For example the following would set the classes on the wrapped DOM node:
 
 ```ts
-import { JSDOM } from 'jsdom';
-import ProjectorMixin from '@dojo/widget-core/mixins/Projector';
-import App from './widgets/App';
-
-const dom = new JSDOM('', { runScripts: 'dangerously' });
-const projector = new (ProjectorMixin(App))();
-const projector.sandbox(dom.window.document);
-
-/* set any App properties/children/state */
-const html = projector.toHtml();
-/* `html` contains the rendered HTML */
+const div = document.createElement('div');
+const WrappedDiv = DomWrapper(div);
+const wNode = w(WrappedDiv, {
+    classes: {
+        'foo': true
+    }
+});
 ```
 
-#### Event Handling
-
-The recommended pattern for custom event handlers is to declare them on the widget class and reference the function using `this`.
-Event handlers are most commonly called from `render`.
-
-Event handlers can be internal logic encapsulated within a widget or delegate to a function passed into the widget via `properties`.
-For convenience event handlers are automatically bound to the scope of their enclosing widget.
+### Meta Configuration
 
-*internally defined handler*
-
-```ts
-class MyWidget extends WidgetBase<WidgetProperties> {
-	private selected: boolean;
+Widget meta is used to access additional information about the widget, usually information only available through the rendered DOM element - for example, the dimensions of an HTML node. You can access and respond to meta data during a widget's render operation.
 
-	onClick() {
-		this.selected = !this.selected;
-	}
+```typescript
+class TestWidget extends WidgetBase<WidgetProperties> {
+    render() {
+        const dimensions = this.meta(Dimensions).get('root');
 
-	render(this: MyWidget): DNode {
-		return v('div', [
-			v('input', { type: 'checkbox', onclick: this.onClick }),
-			v('input', { type: 'text', disabled: this.selected })
-		]);
-	}
+        return v('div', {
+            key: 'root',
+            innerHTML: `Width: ${dimensions.width}`
+        });
+    }
 }
 ```
 
-*Handler passed via properties*
-
-```ts
-class MyWidget extends WidgetBase<WidgetProperties> {
-	onClick(): void {
-		this.properties.mySpecialFunction();
-	}
-}
-```
+If an HTML node is required to calculate the meta information, a sensible default will be returned and your widget will be automatically re-rendered to provide more accurate information.
```

Comment:
Feels like we should add an extra sentence as this is really beneficial, e.g. something like "Meta allows us to preserve the concepts of reactivity in a virtual DOM where traditional vdom approaches would simply provide a reference to a DOM node" but something better?